### PR TITLE
Filter configuration specific files

### PR DESCRIPTION
**Pull request by nadyaA**
Respect prepare/build configuration when preparing project files.

https://github.com/NativeScript/nativescript-cli/issues/2934

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

File: test/unit-tests/mobile/project-files-manager.ts [hunk 1]
```diff
@@ -104,10 +104,18 @@ function createTestInjector(): IInjector {
 }
 
 async function createFiles(testInjector: IInjector, filesToCreate: string[]): Promise<string> {
-	let fs = testInjector.resolve("fs");
 	let directoryPath = temp.mkdirSync("Project Files Manager Tests");
 
-	_.each(filesToCreate, file => fs.writeFile(path.join(directoryPath, file), ""));
+	_.each(filesToCreate, file => createFile(testInjector, file, "", directoryPath));
```
Comment (rosen-vladimirov) [inline test/unit-tests/mobile/project-files-manager.ts]: createFile returns promise, currently noone is awaiting this promise

File: test/unit-tests/mobile/project-files-manager.ts [hunk 2]
```diff
@@ -104,10 +104,18 @@ function createTestInjector(): IInjector {
...
+
+	return directoryPath;
+}
+
+async function createFile(testInjector: IInjector, fileToCreate: string, fileContent: string, directoryPath?: string): Promise<string> {
```
Comment (rosen-vladimirov) [inline test/unit-tests/mobile/project-files-manager.ts]: this method does not have to be async - it doesn't have any async operations in it

File: declarations.d.ts [hunk 3]
```diff
@@ -867,7 +867,7 @@ interface ILiveSyncServiceBase {
 	 * If watch option is not specified executes full sync
 	 * If watch option is specified executes partial sync
 	 */
-	sync(data: ILiveSyncData[], projectId: string, filePaths?: string[]): Promise<void>;
+	sync(data: ILiveSyncData[], projectId: string, filePaths: string[], projectFilesConfig: IProjectFilesConfig): Promise<void>;
```
Comment (rosen-vladimirov) [inline declarations.d.ts]: Why is this changed? The `sync`  method must have `filePaths` as optional argument, as they are passed only when we are not in full sync mode

**Comment by justcodebuilduser** [comment]
:broken_heart:

File: test/unit-tests/mobile/project-files-manager.ts (hunk 1 above)
Comment (nadyaA) [inline test/unit-tests/mobile/project-files-manager.ts]: I changed the method to return string. Thanks

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by nadyaA** [comment]
run ci

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

File: helpers.ts [hunk 4]
```diff
@@ -400,6 +401,13 @@ export async function connectEventuallyUntilTimeout(factory: () => net.Socket, t
 	});
 }
 
+export function getProjectFilesConfig(release: boolean): IProjectFilesConfig {
```
Comment (rosen-vladimirov) [inline helpers.ts]: you can modify this to be:
```TypeScript
export function getProjectFilesConfig(opts: { isReleaseBuild: boolean }): IProjectFilesConfig {
}
```

File: test/unit-tests/mobile/project-files-manager.ts [hunk 5]
```diff
@@ -191,11 +199,39 @@ describe("Project Files Manager Tests", () => {
 		assert.isTrue(fs.exists(path.join(directoryPath, "a.test")));
 	});
 
+	it("filters release specific files", async () => {
+		const directoryPath = await createFile(testInjector, "test.debug.x", "debug");
```
Comment (TomaNikolov) [inline test/unit-tests/mobile/project-files-manager.ts]: Since `createFile` is no longer async you can remove `await`

File: test/unit-tests/mobile/project-files-manager.ts [hunk 6]
```diff
@@ -191,11 +199,39 @@ describe("Project Files Manager Tests", () => {
...
+		const releaseFileContent = "release";
+		await createFile(testInjector, "test.release.x", releaseFileContent, directoryPath);
```
Comment (TomaNikolov) [inline test/unit-tests/mobile/project-files-manager.ts]: Same as above.

File: test/unit-tests/mobile/project-files-manager.ts [hunk 7]
```diff
@@ -191,11 +199,39 @@ describe("Project Files Manager Tests", () => {
...
+
+		const fs = testInjector.resolve("fs");
+		assert.isFalse(fs.exists(path.join(directoryPath, "test.debug.x")));
+		assert.isTrue(fs.exists(path.join(directoryPath, "test.x")));
+		assert.isFalse(fs.exists(path.join(directoryPath, "test.release.x")));
+		assert.isTrue(fs.readFile(path.join(directoryPath, "test.x")).toString() === releaseFileContent);
+	});
+
+	it("filters debug specific files by default", async () => {
+		const directoryPath = await createFile(testInjector, "test.release.x", "release");
```
Comment (TomaNikolov) [inline test/unit-tests/mobile/project-files-manager.ts]: `await`

File: test/unit-tests/mobile/project-files-manager.ts [hunk 8]
```diff
@@ -191,11 +199,39 @@ describe("Project Files Manager Tests", () => {
...
+		const debugFileContent = "debug";
+		await createFile(testInjector, "test.debug.x", debugFileContent, directoryPath);
```
Comment (TomaNikolov) [inline test/unit-tests/mobile/project-files-manager.ts]: `await`

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:
